### PR TITLE
Write dataset description for all converters

### PR DESCRIPTION
**Pull request by ghisvail**
Dataset description files are a requirement for BIDS compliance.

- Define a proper model for the BIDS dataset description
- Ensure the model is used in bids utility functions
- Add missing dataset description for some converters

### CODE REVIEW

**Comment by ghisvail** [comment]
Note that the current definition of the BIDS dataset description model is the most nominal one based on the current specs. But it's implemented in a way that future extension only requires adding a properly typed attribute to the class and a field-to-alias mapping in `Config.fields` if the field is not serialized to pascal case.

File: clinica/iotools/bids_utils.py [hunk 1]
```diff
@@ -1,9 +1,36 @@
...
+
+class BidsDatasetDescription(BaseModel):
+    """Model class for a BIDS description."""
+
+    class Config:
+        fields = {"bids_version": "BIDSVersion"}
+        allow_population_by_field_name = True
+
+        @classmethod
+        def alias_generator(cls, string: str) -> str:
```
Comment (omar-rifai) [inline clinica/iotools/bids_utils.py:19]: Please replace string with a more context-relevant name

File: clinica/iotools/bids_utils.py [hunk 2]
```diff
@@ -1,9 +1,36 @@
...
+
+
+class BidsDatasetDescription(BaseModel):
+    """Model class for a BIDS description."""
+
+    class Config:
+        fields = {"bids_version": "BIDSVersion"}
+        allow_population_by_field_name = True
+
+        @classmethod
```
Comment (omar-rifai) [inline clinica/iotools/bids_utils.py:18]: Shouldn't this be a static method?

File: requirements.txt [hunk 3]
```diff
@@ -24,3 +24,4 @@ click >= 7
 click-option-group >= 0.5, < 0.6
 openpyxl
 fsspec
+pydantic>=1.4
```
Comment (omar-rifai) [inline requirements.txt:27]: In the effort of moving towards typing and making the code more robust, the use of pydantic seems to be relevant.  But please concert before introducing new dependencies to the requirements.

File: clinica/iotools/bids_utils.py [hunk 4]
```diff
@@ -1,9 +1,36 @@
...
 from typing import List
 
+from pydantic import BaseModel
+
+
+class BidsDatasetDescription(BaseModel):
+    """Model class for a BIDS description."""
+
+    class Config:
+        fields = {"bids_version": "BIDSVersion"}
```
[the remark this answers is not in the document]
Comment (ghisvail) [reply clinica/iotools/bids_utils.py:15]: Fixed in #392

**Comment by ghisvail** [comment]
Blocked by #392, which implements a dataset description model for HABS that could be used for the other converters targeted in this PR. Depends which one we want to merge first between #392 and this one.

**Comment by ghisvail** [comment]
We still need to update the CI data to include the missing `dataset_description.json` file for some converters. To be done after the current CI refactoring and testing efforts.

**Comment by ghisvail** [comment]
Before we consider merging this, I'd like to test the same approach with `attrs` (an alternative to `pydantic`), which is used by `pydra` extensively. If we were to consider switching to `pydra` at some point, we might benefit from using the same dependencies.

**Comment by ghisvail** [comment]
I believe the alternative solution in #535 using `attrs` is better from a conceptual and maintenance perspective. No need to learn the intricacies of `pydantic` (especially the `Config` inner class) and closer to the composition over inheritence principle where Python has been going lately. The `attrs` solution is basically a `dataclass` (record type) composed with a converter registry provided by `cattrs`. Simple.

**Comment by omar-rifai** [comment]
Should we close this PR?

**Comment by ghisvail** [comment]
It will be closed automatically once #535 is merged

**Pull request closed without merging**